src/Parser.php

```diff
@@ -87,8 +86,11 @@ public function isHost(string $host): bool
      */
     public function isPort($port): bool
     {
-        return in_array($port, [null, ''], true)
-            || filter_var($port, FILTER_VALIDATE_INT, ['options' => ['min_range' => 1, 'max_range' => 65535]]);
+        if (null === $port || '' === $port) {
+            return true;
+        }
+
+        return (bool) filter_var($port, FILTER_VALIDATE_INT, ['options' => ['min_range' => 1, 'max_range' => 65535]]);
```

Comment:
why casting to boolean ? the conditions states the port must be at least 1 so any valid value will return a positive integer thus `true` ?

Reply:
With strict types, a `TypeError` is thrown if the value returned isn't explicitly a boolean.